lib/project.ts

```diff
@@ -234,9 +235,7 @@ export default class Project {
         lint: this.lint,
         audit: this.audit
       });
-      // TODO: revisit to fix promise issues
-      // tslint:disable-next-line:no-floating-promises
-      addonProject.watch({ onBuild: options.onBuild, outputDir: addonDist });
+      return addonProject.watch({ onBuild: options.onBuild, outputDir: addonDist });
```

Comment:
It does return a promise, and it makes it so the watchers are started serially (i.e. we don't have two watchers of child addons starting at the same time). As to whether making the watchers start serially actually _does_ anything I'm not sure.